orangewidget/utils/itemmodels.py

```diff
@@ -218,10 +211,25 @@ def sort(self, column: int, order: Qt.SortOrder = Qt.AscendingOrder):
             persistent,
             [self.index(row, pind.column())
              for row, pind in zip(persistent_rows, persistent)])
-        if QT_VERSION >= 0x50000:
-            self.layoutChanged.emit([], QAbstractTableModel.VerticalSortHint)
-        else:
-            self.layoutChanged.emit()
+        self.layoutChanged.emit([], QAbstractTableModel.VerticalSortHint)
+
+    def _sort(self, column, order):
+        indices = None
+        if column >= 0:
+            data = numpy.asarray(self._sortColumnData(column))
+            if data is None:
+                data = numpy.arange(self.rowCount())
+            elif data.dtype == object:
+                data = data.astype(str)
+
+            # _sortColumnData returns data in its currently shown order
+            # _argSortData thus returns an array a, in which a[i] is the row
+            # number (in the current view) to put to line i
+            # mapToSourceRows maps these indices back to original data
+            # @kernc, tbanks for this joke. :) Couldn't _sortColumnData simply
+            # return the original data, saving us from double mapping?
```

Comment:
No. This is done to ensure stable sort order on consecutive calls.

Reply:
Thanks! I'll document that. Seeing the code, I had to convince myself it works at all, then I saw the trick but didn't understand why.